### PR TITLE
Devops-14 Change ci of avn-parachain

### PR DESCRIPTION
**Pull request by lbaiao2019**
# Thank you for your Pull Request

AFTER YOU'VE READ THIS NOTICE, FEEL FREE TO REMOVE IT.
Thank you!

Before you submit a PR, please check that:

- [ ] You describe the purpose of the PR, e.g.:
  - What does it do?
  - Highlight what important points reviewers should know about;
  - Indicates if there is something left for follow-up PRs.
- [ ] Documentation updated
- [ ] Business logic tested successfully
- [ ] Verify First, Write Last: In Substrate development, it is important that you always ensure preconditions are met and return errors at the beginning. After these checks have completed, then you may begin the function's computation.

✄ -----------------------------------------------------------------------------

### CODE REVIEW

**Review by RuiSMagalhaes — APPROVED** [review]
I like the approach 💯

**Comment by RuiSMagalhaes** [comment]
can we include @nahuseyoum and @thadouk to this one?

**Review by thadouk — COMMENTED** [review]
It isn't clear what the change does exactly, a better description in the PR will help.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -44,11 +44,8 @@ jobs:
       - name: Linting check
         run: cargo fmt --all --check
 
-      - name: Check code
-        run: cargo check --locked --release
-
-      - name: Test Code
-        run: cargo test
+      - name: Check code & Test Code
+        run: cargo check --locked --release & cargo test & wait
```
[the remark this answers is not in the document]
Comment (lbaiao2019) [reply .github/workflows/ci.yml:48]: The task is to run both stages in parallel. https://github.com/Aventus-Network-Services/avn-parachain/blob/main/.github/workflows/ci.yml#L47-L51

no make sense to create a complex workflow with a matrix to do that.

**Review by vukomir — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -38,17 +38,65 @@ jobs:
...
 
-      #- name: benchmark test Code
-      #  run: cargo test --features runtime-benchmarks -- benchmarks
\ No newline at end of file
+      - name: Restore cargo cache - debug
+        uses: actions/cache@v3
+        with:
+          path: |
+            target/
+          key: ${{ runner.os }}-cargo-release-${{ hashFiles('**/Cargo.lock') }}
```
Comment (thadouk) [inline .github/workflows/ci.yml:99]:
```suggestion
          key: ${{ runner.os }}-cargo-debug-${{ hashFiles('**/Cargo.lock') }}      
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -38,17 +38,55 @@ jobs:
...
         run: cargo fmt --all --check
 
+  check:
+    name: Check code
+    runs-on: avn-runner    
+    needs: ci
+    timeout-minutes: 90
+    steps:
+      - uses: actions/checkout@v3
+
```
Comment (thadouk) [inline .github/workflows/ci.yml:55]:
```suggestion
      - name: Restore cargo cache - common
        uses: actions/cache@v3
        with:
          path: |
            ~/.cargo/bin/
            ~/.cargo/registry/index/
            ~/.cargo/registry/cache/
            ~/.cargo/git/db/
          key: ${{ runner.os }}-cargo-common-${{ hashFiles('**/Cargo.lock') }}
```

**Review by thadouk — APPROVED** [review]

**Review by nahuseyoum — APPROVED** [review]

**Review by vukomir — APPROVED** [review]

**Review by RuiSMagalhaes — APPROVED** [review]

**Pull request merged**